File: InternetDemoJava/app/src/main/java/io/mastercoding/internetdemojava/MainActivity.java
package io.mastercoding.internetdemojava;

import androidx.activity.OnBackPressedCallback;
import androidx.appcompat.app.AppCompatActivity;

import android.net.ConnectivityManager;
import android.net.Network;
import android.net.NetworkCapabilities;
import android.os.Bundle;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private WebView webView;
    private OnBackPressedCallback webViewBackCallback;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        webView = findViewById(R.id.webView);

        webView.setWebViewClient(new WebViewClient());

        WebSettings settings = webView.getSettings();
        settings.setJavaScriptEnabled(true);
        settings.setDomStorageEnabled(true);
        settings.setLoadWithOverviewMode(true);
        settings.setUseWideViewPort(true);

        if (isOnline()) {
            webView.loadUrl("https://www.google.com");
        } else {
            Toast.makeText(this, "No internet connection", Toast.LENGTH_LONG).show();
        }

        // Recommended back handling with dispatcher
        webViewBackCallback = new OnBackPressedCallback(true) {
            @Override
            public void handleOnBackPressed() {
                if (webView != null && webView.canGoBack()) {
                    webView.goBack();
                } else {
                    setEnabled(false);
                    MainActivity.super.onBackPressed();
                }
            }
        };
        getOnBackPressedDispatcher().addCallback(this, webViewBackCallback);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (webViewBackCallback != null) {
            webViewBackCallback.remove();
        }
    }

    // 👇🏽 Make sure this method is INSIDE the class, not after the final }
    private boolean isOnline() {
        ConnectivityManager cm = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
        if (cm == null) return false;

        Network active = cm.getActiveNetwork();
        if (active == null) return false;

        NetworkCapabilities caps = cm.getNetworkCapabilities(active);
        if (caps == null) return false;

        return caps.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) &&
                (caps.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
                        || caps.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)
                        || caps.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET));
    }
} // 👈🏽 final closing brace of the class — nothing should be after this

File: Broadcast_Receiver_Intent_Filter/app/src/main/java/io/mastercoding/broadcast_receiver_intent_filter/MainActivity.kt
package io.mastercoding.broadcast_receiver_intent_filter

import android.content.IntentFilter
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    lateinit var airplanemodeReicever: AirphoneModeReicever
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)


        var intentfilter: IntentFilter= IntentFilter("android.intent.action.AIRPLANE_MODE")
         airplanemodeReicever= AirphoneModeReicever()
        registerReceiver(airplanemodeReicever,intentfilter)

    }

    override fun onDestroy() {
        unregisterReceiver(airplanemodeReicever)
        super.onDestroy()
    }
}

